gbpservice/nfp/configurator/agents/loadbalancer_v2.py

```diff
@@ -0,0 +1,979 @@
+
+    def delete_loadbalancer(self, context, loadbalancer):
+        """Enqueues event for worker to process delete loadbalancer request.
+
+        :param context: RPC context
+        :param loadbalancer: loadbalancer resource to be deleted
+
+        Returns: None
+
+        """
+        arg_dict = {'context': context,
+                    'loadbalancer': loadbalancer,
+                    }
+        self._send_event(lb_constants.EVENT_DELETE_LOADBALANCER, arg_dict,
+                         serialize=True, binding_key=loadbalancer['id'],
+                         key=loadbalancer['id'])
+
+    def create_listener(self, context, listener):
+        """Enqueues event for worker to process create listener request.
+
+        :param context: RPC context
+        :param listener: listener resource to be created
+
+        Returns: None
+
+        """
+        arg_dict = {'context': context,
+                    'listener': listener,
+                    }
+        self._send_event(lb_constants.EVENT_CREATE_LISTENER, arg_dict,
+                         serialize=True,
+                         binding_key=listener['loadbalancer_id'],
+                         key=listener['id'])
+
+    def update_listener(self, context, old_listener, listener):
+        """Enqueues event for worker to process update listener request.
+
+        :param context: RPC context
+        :param old_listener: old listener resource to be updated
+        :param listener: new listener resource
+
+        Returns: None
+
+        """
+        arg_dict = {'context': context,
+                    'old_listener': old_listener,
+                    'listener': listener,
+                    }
+        self._send_event(lb_constants.EVENT_UPDATE_LISTENER, arg_dict,
+                         serialize=True,
+                         binding_key=listener['loadbalancer_id'],
+                         key=listener['id'])
+
+    def delete_listener(self, context, listener):
+        """Enqueues event for worker to process delete listener request.
+
+        :param context: RPC context
+        :param listener: listener resource to be deleted
+
+        Returns: None
+
+        """
+        arg_dict = {'context': context,
+                    'listener': listener,
+                    }
+        self._send_event(lb_constants.EVENT_DELETE_LISTENER, arg_dict,
+                         serialize=True,
+                         binding_key=listener['loadbalancer_id'],
+                         key=listener['id'])
+
+    def create_pool(self, context, pool):
+        """Enqueues event for worker to process create pool request.
+
+        :param context: RPC context
+        :param pool: pool resource to be created
+
+        Returns: None
+
+        """
+        arg_dict = {'context': context,
+                    'pool': pool
+                    }
+        # TODO: M:N pool is not yet implemented.
+        # For Mitaka, use binding_key=pool['listeners'][0]['id']
```

Comment:
@njagadish this is an outdated comment. The bind_key should always be the corresponding loadbalancer.